integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,1049 @@
+	updatedConfig := proto.Clone(currentConfig).(*common.Config)
+	updatedConfig.ChannelGroup.Groups["Application"].Groups[org.Name].Policies["Writers"].Policy.Value = protoutil.MarshalOrPanic(policydsl.SignedByMspMember(org.MSPID))
+	nwo.UpdateConfig(n, orderer, channelID, currentConfig, updatedConfig, true, peer)
+
+	By("joining a new channel testchannelID3 on peer " + peerID)
+	n.JoinChannel(testchannelID3, orderer, peer)
+
+	By("waiting for all peers to have the same ledger height")
+	channelPeers := n.PeersWithChannel(channelID)
+	maxLedgerHeight := nwo.GetMaxLedgerHeight(n, channelID, channelPeers...)
+	nwo.WaitUntilEqualLedgerHeight(n, channelID, maxLedgerHeight, channelPeers...)
+
+	helper.invokeMarblesChaincode(chaincode.Name, peer, "transferMarble", testKey, "newowner_update_"+newCCSequence)
+
+	By("verifying qscc GetBlockByNumber returns an error for block number before snapshot on peer " + peerID)
+	resp := callQSCC(n, peer, "qscc", "GetBlockByNumber", 1, "testchannel", strconv.Itoa(lastBlockInSnapshot))
+	Expect(resp).To(ContainSubstring(fmt.Sprintf("The ledger is bootstrapped from a snapshot. First available block = [%d]", lastBlockInSnapshot+1)))
+
+	By("verifying qscc GetBlockByTxID returns an error for old txid before snapshot on peer " + peerID)
+	resp = callQSCC(n, peer, "qscc", "GetBlockByTxID", 1, "testchannel", oldTxid)
+	Expect(resp).To(ContainSubstring(fmt.Sprintf("Failed to get block for txID %s, error details for the TXID [%s] not available. Ledger bootstrapped from a snapshot. First available block = [%d]",
+		oldTxid, oldTxid, lastBlockInSnapshot+1)))
+
+	By("verifying qscc GetTransactionByID returns an error for old txid before snapshot on peer " + peerID)
+	resp = callQSCC(n, peer, "qscc", "GetTransactionByID", 1, "testchannel", oldTxid)
+	Expect(resp).To(ContainSubstring(fmt.Sprintf("Failed to get transaction with id %s, error details for the TXID [%s] not available. Ledger bootstrapped from a snapshot. First available block = [%d]",
+		oldTxid, oldTxid, lastBlockInSnapshot+1)))
+
+	By("verifying qscc GetBlockByTxID succeeds for a txid after snapshot on peer " + peerID)
+	newTxid := getTxIDFromLastBlock(n, peer)
+	callQSCC(n, peer, "qscc", "GetBlockByTxID", 0, "testchannel", newTxid)
+
+	By("verifying qscc GetTransactionByID succeeds for a txid after snapshot on peer " + peerID)
+	callQSCC(n, peer, "qscc", "GetTransactionByID", 0, "testchannel", newTxid)
+
+	By("verifying qscc GetBlockByNumber succeeds for a block number after snapshot on peer " + peerID)
+	callQSCC(n, peer, "qscc", "GetBlockByNumber", 0, "testchannel", fmt.Sprintf("%d", lastBlockInSnapshot+1))
+
+	By("verifying history on peer " + peerID)
+	By("verifying history query")
+	expectedHistoryResult := []*marbleHistoryResult{
+		{IsDelete: "false", Value: newMarble(testKey, "blue", 35, "newowner_update_"+newCCSequence)},
+		{IsDelete: "false", Value: newMarble(testKey, "blue", 35, "newowner_"+newCCSequence)},
+	}
```

Comment:
this is very hidden here in the helper function...  the expected history should be passed as an parameter to this function.